scripts/UnitTests/Backend.Tests.ps1

```diff
@@ -16,3 +16,97 @@ Describe "Test Hyper-V Backend instance" {
         Assert-VerifiableMocks
     }
 }
+
+Describe "Test Azure Backend instance" {
+    Mock Write-Verbose -Verifiable {return}
+    Mock login_azure -Verifiable {return}
+    #Mock Write-Warning  -MockCalled {return}
+    Mock Stop-AzureRmVM -Verifiable {return}
+    Mock Remove-AzureRmVM -Verifiable {return}
+    Mock Get-AzureRmNetworkSecurityGroup -Verifiable {return [Microsoft.Azure.Commands.Network.Models.PSNetworkSecurityGroup]::new() }
+  #  Mock New-AzureRmNetworkSecurityRuleConfig -Verifiable {return  [Microsoft.Azure.Commands.Network.Models.PSSecurityRule]::new()}
+  #  Mock New-AzureRmNetworkSecurityGroup -Verifiable {return}
+    Mock Get-AzureRmPublicIpAddress -Verifiable {return [Microsoft.Azure.Commands.Network.Models.PSPublicIpAddress]::new()}
+  #  Mock New-AzureRmPublicIpAddress -Verifiable {return}
+    Mock Get-AzureRmVirtualNetwork -Verifiable {return [Microsoft.Azure.Commands.Network.Models.PSVirtualNetwork]::new() }
+    Mock Get-AzureRmVirtualNetworkSubnetConfig -Verifiable {return [Microsoft.Azure.Commands.Network.Models.PSSubnet]::new()}
+    Mock New-AzureRmVMConfig -Verifiable {return [Microsoft.Azure.Commands.Compute.Models.PSVirtualMachine]::new()}
+    Mock Get-AzureRmNetworkInterface -Verifiable {return [Microsoft.Azure.Commands.Network.Models.PSNetworkInterface]::new()}
+    Mock Set-AzureRmNetworkInterface -Verifiable {return [Microsoft.Azure.Commands.Network.Models.PSNetworkInterface]::new()}
+    #Mock New-AzureRmNetworkInterface -Verifiable {return}
+    Mock Add-AzureRmVMNetworkInterface -Verifiable {return [Microsoft.Azure.Commands.Compute.Models.PSVirtualMachine]::new()}
+    Mock Set-AzureRmVMOSDisk -Verifiable {return [Microsoft.Azure.Commands.Compute.Models.PSVirtualMachine]::new()}
+    Mock New-AzureRmVM -Verifiable {return [Microsoft.Azure.Commands.Compute.Models.PSAzureOperationResponse]::new()}
+  #  Mock New-AzureRmVirtualNetwork -Verifiable {return }
+#    Mock New-AzureRmVirtualNetworkSubnetConfig -Verifiable {return}
+#    Mock Add-AzureRmVirtualNetworkSubnetConfig -Verifiable {return}
+  #  Mock Set-AzureRmVirtualNetwork -Verifiable {return }
+    <#
+    Mock Remove-AzureRmNetworkInterface -Verifiable {return}
+    Mock Remove-AzureRmPublicIpAddress -Verifiable {return}
+    Mock New-AzureRmResource -Verifiable {return}#>
+    $backendFactory = [BackendFactory]::new()
+    It "Should test Backend Factory " {
+        $backendFactory | Should Not Be $null
+    }
+    $azureBackend = $backendFactory.GetBackend("AzureBackend",@(1))
+
+    $fName = "test"
+    $fuseIPW = "yes"
+    $fimageName = "123456789012345678901234567890123"
```

Comment:
when you write tests, it is a convention to define strings starting with fake_test or fake_image_name